feeds/product.php

```diff
@@ -38,6 +38,151 @@
 
 DfTools::validateSecurityToken(Tools::getValue('dfsec_hash'));
 
+/**
+ * Global storage for all active measurements.
+ *   - $__metrics_data[$label] holds ['time' => float, 'memory' => int, 'peak' => int]
+ *   - $__metrics_stack is an indexed array of labels, used to pop the most recent.
+ */
+$__metrics_data  = [];
+$__metrics_stack = [];
+
+/**
+ * Begin measuring time & memory.
+ *
+ * @param string|null $label  Optional: a custom label for this measurement.
+ *                            If you pass null (or omit), a unique label is generated.
+ * @return string             The label that was used.  (Use the same label in stop_metrics() if you want.)
+ */
+function start_metrics(string $label = null): string
+{
+    global $__metrics_data, $__metrics_stack;
+
+    // 1) If no label provided, generate one
+    if ($label === null) {
```

Comment:
Yoda conditions are preferred:

```suggestion
    if (null === $label) {
```